src/components/IndividualArticleCard.jsx

```diff
@@ -1,14 +1,41 @@
+import { useEffect, useState } from 'react';
+import { updateVotes } from '../utils/api';
+
 const IndividualArticleCard = ({ article }) => {
+  const [votes, setVotes] = useState(article.votes);
+  const [hasVoted, setHasVoted] = useState(false);
+  const [newVote, setNewVote] = useState(0);
+  const [err, setErr] = useState(null);
+  const { article_id } = article;
+
+  const handleClick = (num) => {
+    setNewVote((currVotes) => currVotes + num);
+    setHasVoted(true);
+  };
+
+  useEffect(() => {
```

Comment:
I'm not sure that we need a useEffect here. The logic can all be handled in the handleClick function as we can update state from there and there. This should also include an optimistic render of the vote ie 

- update UI
- make api call
- reverse UI change if an error is caught